Clean up files that were created previously for the sidebar metabox that was overwritten by Steve's work in #72

zainabD opened the pull request:
I added a new Javascript file called **admin-blocks.js**, enqueued it in admin side, so that we can place all block related functions in there (functions like register block, unregister block, render block, enqueue assets and etc

Changes proposed in this pull request:

- A new .js file called **admin-blocks.js**, where I unregistered 3 blocks per request [#81](https://github.com/asu-ke-web-services/UDS-WordPress-Theme/issues/81) 
- Updated **inc/enqueue.php**, called the new js file in admin side using admin_enqueue_scripts action, and deleted the call of metaboxes.js, since these functions were used for the sidebar metabox that was overwritten by Steve's work in [#72](https://github.com/asu-ke-web-services/UDS-WordPress-Theme/pull/72).
- Deleted the Metaboxes.js for same reason that was mentioned above.

wmcconnell commented on inc/enqueue.php:
```diff
@@ -67,11 +67,11 @@ function uds_wp_metaboxes_scripts() {
 		if ( 'post.php' == $pagenow || 'post-new.php' == $pagenow ) {
 
-			wp_enqueue_script( 'uds-wordpress-sidebar-metabox-scripts', get_template_directory_uri() . '/js/metaboxes.js', array( 'jquery' ), $js_metaboxes_version, true );
+			wp_enqueue_script( 'uds-wordpress-sidebar-metabox-scripts', get_template_directory_uri() . '/js/admin-blocks.js', array( 'jquery' ), $js_admin_blocks_version, true );
```
It looks like we left the original file handle of 'uds-wordpress-sidebar-metabox-scripts' in there. While it would still work (as long as that name is unique), we should probably rename this to avoid confusion since this is not a script about meta boxes. I would suggest something like 'uds-wordpress-admin-blocks' to replace that here, and in line 75 below.

wmcconnell commented on inc/enqueue.php:
```diff
@@ -67,11 +67,11 @@ function uds_wp_metaboxes_scripts() {
 
-			wp_enqueue_script( 'uds-wordpress-sidebar-metabox-scripts', get_template_directory_uri() . '/js/metaboxes.js', array( 'jquery' ), $js_metaboxes_version, true );
+			wp_enqueue_script( 'uds-wordpress-sidebar-metabox-scripts', get_template_directory_uri() . '/js/admin-blocks.js', array( 'jquery' ), $js_admin_blocks_version, true );
 			wp_enqueue_style( 'uds-wordpress-sidebar-metabox-styles', get_template_directory_uri() . '/css/admin.min.css', array(), $css_metaboxes_version );
```
Just a reminder to use the same script handle from line 74 above, once we have changed it to better match what this script actually does.

zainabD replied:
This css file only has the metabox related styles, so I'm not sure if we need to rename it or to just delete the file since it's not being used at the time.
What do you think?

wmcconnell reviewed (changes requested):
Could we rename the handle for this new script, per my comments? Since this is not actually a script that deals with meta box styles, we should use a handle that matches what the script is doing - just to avoid confusion.

zainabD reviewed (commented):
I updated the function name and the new script name but wasn't sure about the css file since it only has the metaboxes related styles.

SteveRyan-ASU commented:
Man, did this open up a can of worms on my end... totally learned some stuff about registering and unregistering blocks that I didn't know before. 🤷‍♂️ 

Pertaining to the work done against the requirement from #81 to remove blocks, here's what I am wondering about:

### If we think we will get into a situation in which we would want to restrict blocks to a specific page template, then the JS required gets immediately more complicated.

- See this article for the full description of why: https://webservices.ufhealth.org/2020/07/08/how-to-restrict-blocks-to-specific-page-templates-in-the-wordpress-gutenberg-editor/
- TL;DR: Since the page template can be switched "on the fly" from within the editor, there are LOTS of things we should potentially do to check the state of the allowed/not-allowed blocks upon the user clicking the dropdown.

### The process for unregistering blocks from the editor via PHP appears a lot easier. However, there is a catch.

- Here's the filter: https://developer.wordpress.org/reference/hooks/allowed_block_types/
- The second parameter of that filter also allows you to reference the allowed block types list against a specific `wp_post` object.
- The `wp_post` object contains references to individual pages, page slugs or specific post types. But not to the page template within the same post type.

### What I'm wondering is the following:

- Should we consider moving the `landing.php` template out of the native post type `pages`.
- ... and instead, create a custom post type called `Landing` with the same intent & markup as the landing.php template.
- ... so that this whole process of limiting blocks to specific regions of WordPress becomes easier.

zainabD commented:
Per our last discussion with Steve and Walter, we decided to handle the block registration functions using php instead of JS, therefore we don't need the js file that was created in this branch, we only need to remove all the remainings of the sidebar metabox from old branch (this was overwritten by Steve's work in #72).
- I deleted all files, functions and classes that were created previously for the sidebar metabox that was overwritten by Steve's work in #72.
- Deleted **js/metaboxes.js** 
- Deleted function '**uds_wp_metaboxes_scripts**' from **inc/enqueue.php** 
- Deleted css class that were created for the sidebar metabox from **css/admin.css**

wmcconnell reviewed (approved):
Looks good. Approved!
